node/src/components/small_network.rs

```diff
@@ -638,29 +638,39 @@ where
     ) -> Effects<Event<P>> {
         let _ = self.pending.remove(&peer_address);
 
-        let mut effects = Effects::new();
-
         if let Some(peer_id) = peer_id {
             if let Some(ref err) = error {
-                warn!(our_id=%self.our_id, %peer_id, %peer_address, %err, "outgoing connection failed");
+                warn!(
+                    our_id=%self.our_id,
+                    %peer_id,
+                    %peer_address,
+                    %err,
+                    "outgoing connection failed"
+                );
             } else {
                 warn!(our_id=%self.our_id, %peer_id, %peer_address, "outgoing connection closed");
             }
-            effects.extend(self.remove(effect_builder, &peer_id, false));
-        } else {
-            // If we are not calling remove, call the reconnection check explicitly.
-            effects.extend(self.reconnect_if_not_connected_to_any_known_addresses(effect_builder));
+            return self.remove(effect_builder, &peer_id, false);
         }
 
         // If we don't have the node ID passed in here, it was never added as an
         // outgoing connection, hence no need to call `self.remove()`.
         if let Some(ref err) = error {
-            warn!(our_id=%self.our_id, %peer_address, %err, "outgoing connection failed");
+            warn!(
+                our_id=%self.our_id,
+                %peer_address,
+                %err,
+                "outgoing connection to known address failed"
+            );
         } else {
-            warn!(our_id=%self.our_id, %peer_address, "outgoing connection closed");
+            warn!(
+                our_id=%self.our_id,
+                %peer_address,
+                "outgoing connection to known address closed"
+            );
         }
-
-        effects
+        // Since we are not calling `self.remove()`, call the reconnection check explicitly.
+        self.reconnect_if_not_connected_to_any_known_addresses(effect_builder)
```

Comment:
Would it be possible to rewrite this whole function to get rid of early return from above? By the time I am reading _this_ comment I already lost the context and information form the earlier `if`s and `else`s and have no idea why we're not calling `self.remove()`, why we should have or what would it have meant.

Reply:
I actually just changed it to have an early return, since that makes it simpler imo.  I'd rather not change it back.  I tend to prefer early returns as they generally make the code simpler to understand and require less nested levels.

The diff is pretty hard to read, but looking at [the actual function itself](https://github.com/CasperLabs/casper-node/blob/da770ea81b8ac121bb9db9fb50789d7cb552d4e7/node/src/components/small_network.rs#L632-L674) should hopefully answer your concerns.  It's a fairly small, succinct method really.